File: server/matchmaking/src/main/java/me/eddiep/ghost/matchmaking/network/packets/SetNamePacket.java
package me.eddiep.ghost.matchmaking.network.packets;

import me.eddiep.ghost.matchmaking.Main;
import me.eddiep.ghost.matchmaking.network.PlayerClient;
import me.eddiep.ghost.matchmaking.network.TcpServer;
import me.eddiep.ghost.network.packet.Packet;
import me.eddiep.ghost.network.validate.DummyValidator;

import java.io.IOException;

public class SetNamePacket extends Packet<TcpServer, PlayerClient> {
    public SetNamePacket(PlayerClient client, byte[] data) {
        super(client, data);
    }

    @Override
    public void onHandlePacket(PlayerClient client)  throws IOException {
        String name = consume(255).asString();

        if (Main.SESSION_VALIDATOR instanceof DummyValidator) {
            client.getPlayer().setName(name.trim());
        }
    }
}

File: client/core/src/me/eddiep/ghost/client/network/packets/SetNamePacket.kt
package me.eddiep.ghost.client.network.packets

import me.eddiep.ghost.client.network.Packet
import me.eddiep.ghost.client.network.PlayerClient

class SetNamePacket : Packet<PlayerClient>() {

    override fun write(vararg args : Any) {
        var username = args[0] as String

        for (i in username.length..254) {
            username += " "
        }

        write(0x41.toByte())
        write(username)
        endTCP()
    }
}
